src/prefect/task_engine.py

```diff
@@ -1262,12 +1274,12 @@ async def initialize_run(
                                 wait_for=self.wait_for,
                                 extra_task_inputs=dependencies,
                             )
-                    # Emit an event to capture that the task run was in the `PENDING` state.
-                    self._last_event = emit_task_run_state_change_event(
-                        task_run=self.task_run,
-                        initial_state=None,
-                        validated_state=self.task_run.state,
-                    )
+                            # Emit an event to capture that the task run was in the `PENDING` state.
+                            self._last_event = emit_task_run_state_change_event(
```

Comment:
I'm waiting to consolidate until after client side orchestration is enabled, so we can just cull the else blocks